tools/script_runner.py

```diff
@@ -33,6 +33,26 @@ def reversedate() -> str:
     formatted_date = current_date.strftime("%Y%m%d")
     return formatted_date
 
+def estimate_size_lmdb(file_name: str) -> int:
+    # Check if the file exists
+    if not os.path.isfile(file_name):
+        log_all(logging.ERROR, f'file {file_name}does not exist')
+        raise
+
+    # Get the size of the file in bytes
+    file_size = os.path.getsize(file_name)
+
+    # Calculate the estimated size based on the file factor
+    if file_name.endswith(".zip") or file_name.endswith(".gzip"):
```

Comment:
Wouldn't it be better to list the _contents_ (not the compressed contents) of the zip / gzip file and find the ratio? 

```
import zipfile

def get_zip_uncompressed_size(zip_path):
    with zipfile.ZipFile(zip_path, 'r') as zf:
        return sum(info.file_size for info in zf.infolist())

zip_size = get_zip_uncompressed_size("example.zip")
print(f"Uncompressed ZIP size: {zip_size} bytes")
```

For gzip it works up to 4GB.
```
import struct

def get_gzip_uncompressed_size(gzip_path):
    with open(gzip_path, "rb") as f:
        f.seek(-4, 2)  # Seek to last 4 bytes
        return struct.unpack("<I", f.read(4))[0]  # Little-endian unsigned int

gzip_size = get_gzip_uncompressed_size("example.gz")
print(f"Uncompressed GZIP size: {gzip_size} bytes")
```